test/src/in_language_tests/test_programs/bytes_inline_tests/src/splice.sw

```diff
@@ -0,0 +1,186 @@
+    let _spliced = bytes.splice(2, 1, Bytes::new());
+}
+
+#[test(should_revert)]
+fn revert_bytes_splice_end_out_of_bounds() {
+    let (mut bytes, _, _, _) = setup();
+    let _spliced = bytes.splice(0, bytes.len() + 1, Bytes::new());
+}
+
+/// Additional tests for replacing a spliced range with different Byte lengths.
+#[test()]
+fn bytes_splice_replace_smaller() {
+    let (mut bytes, a, b, c) = setup();
+    bytes.push(11u8);
+    bytes.push(13u8);
+    // bytes = [5, 7, 9, 11, 13]
+    let mut replacement = Bytes::new();
+    replacement.push(42u8);
+    // Remove [1..4) => [7, 9, 11], replace with [42]
+    let spliced = bytes.splice(1, 4, replacement);
+
+    // bytes => [5, 42, 13]
+    assert(bytes.len() == 3);
+    assert(bytes.get(0).unwrap() == a);
+    assert(bytes.get(1).unwrap() == 42u8);
+    assert(bytes.get(2).unwrap() == 13u8);
+
+    // spliced => [7, 9, 11]
+    assert(spliced.len() == 3);
+    assert(spliced.get(0).unwrap() == b);
+    assert(spliced.get(1).unwrap() == c);
+    assert(spliced.get(2).unwrap() == 11u8);
+}
+
+#[test()]
+fn bytes_splice_replace_larger() {
+    let (mut bytes, a, b, c) = setup();
+    // bytes = [5, 7, 9]
+    let mut replacement = Bytes::new();
+    replacement.push(42u8);
+    replacement.push(50u8);
+    replacement.push(60u8);
+    // Remove [1..2) => [7], replace with [42, 50, 60]
+    let spliced = bytes.splice(1, 2, replacement);
+
+    // bytes => [5, 42, 50, 60, 9]
+    assert(bytes.len() == 5);
+    assert(bytes.get(0).unwrap() == a);
+    assert(bytes.get(1).unwrap() == 42u8);
+    assert(bytes.get(2).unwrap() == 50u8);
+    assert(bytes.get(3).unwrap() == 60u8);
+    assert(bytes.get(4).unwrap() == c);
+
+    // spliced => [7]
+    assert(spliced.len() == 1);
+    assert(spliced.get(0).unwrap() == b);
+}
+
+#[test()]
+fn bytes_splice_replace_same_length() {
+    let (mut bytes, a, b, c) = setup();
+    // bytes = [5, 7, 9]
+    let mut replacement = Bytes::new();
+    replacement.push(42u8);
+    replacement.push(50u8);
+    // Remove [1..3) => [7, 9], replace with [42, 50] (same length = 2)
+    let spliced = bytes.splice(1, 3, replacement);
+
+    // bytes => [5, 42, 50]
+    assert(bytes.len() == 3);
+    assert(bytes.get(0).unwrap() == a);
+    assert(bytes.get(1).unwrap() == 42u8);
+    assert(bytes.get(2).unwrap() == 50u8);
+
+    // spliced => [7, 9]
+    assert(spliced.len() == 2);
+    assert(spliced.get(0).unwrap() == b);
+    assert(spliced.get(1).unwrap() == c);
+}
+
+#[test()]
+fn bytes_splice_replace_empty_bytes() {
+    let (mut bytes, a, b, c) = setup();
+    // bytes = [5, 7, 9]
+    let replacement = Bytes::new();
+    // Remove [0..1) => [5], replace with []
+    let spliced = bytes.splice(0, 1, replacement);
+
+    // bytes => [7, 9]
+    assert(bytes.len() == 2);
+    assert(bytes.get(0).unwrap() == b);
+    assert(bytes.get(1).unwrap() == c);
+
+    // spliced => [5]
+    assert(spliced.len() == 1);
+    assert(spliced.get(0).unwrap() == a);
+}
```

Comment:
This case additonally tests one of the  situations where the end of the original buffer overlaps with the end of the new buffer, i.e., the case where you need the temporary buffer.

You should probably add a test for the other case, i.e, 

```
// bytes = [5, 7, 9]
// Remove [0..1] => [5], replace with [10, 12]
```